### PR TITLE
fix(io-sign): [SFEQS-2089] Fix x-io-sign-environment header decode

### PR DESCRIPTION
**Pull request by lucacavallaro**
#### List of Changes

Add a type guard to work with `req.headers` with the right runtime type (`Map` instead of `Record<string, string>`)

#### Motivation and Context

Due to a type annotation error on the generated api client, `getEnvironmentFromHeaders` wrongly assumed that `req.headers` was a `Record` instead of an `Headers` map, so the decode step has always failed.

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/services/ioSignService.ts [hunk 1]
```diff
@@ -57,7 +57,11 @@ const userNotFound =
   "The user associated with this profile could not be found.";
 
 export const getEnvironmentFromHeaders = flow(
-  lookup("x-io-sign-environment"),
+  O.fromPredicate(
+    (u: unknown): u is NodeFetchHeaders => u instanceof NodeFetchHeaders
```
Comment (gunzip) [inline src/services/ioSignService.ts:61]:
```suggestion
    (headers: unknown): headers is NodeFetchHeaders => headers instanceof NodeFetchHeaders
```

or `h`

File: src/services/ioSignService.ts [hunk 2]
```diff
@@ -57,7 +57,12 @@ const userNotFound =
   "The user associated with this profile could not be found.";
 
 export const getEnvironmentFromHeaders = flow(
```
Comment (gunzip) [inline src/services/ioSignService.ts:59]: what if our fetch implementation changes?

Comment (lucacavallaro) [reply src/services/ioSignService.ts:59]: if our fetch implementation changes we will have giant problems all over the project :) the autogenerated client is not fetch-agnostic

Comment (gunzip) [reply src/services/ioSignService.ts:59]: can unit tests catch it?

Comment (lucacavallaro) [reply src/services/ioSignService.ts:59]: if there are used other fetch implementation this method will return `prod` (the fallback value), there are no crashes or unhandled behavior

**Pull request merged**